### ISSUE TITLE
🐛 [Bug] `aten.convolution` fails when 1D stride is tuple

### ISSUE THREAD

**Issue by gs-olive** [pytorch/TensorRT#2185]
##  Bug Description

When encountering a Conv1D operator in Dynamo, PyTorch can switch integer components, like dilation, into tuples. This is problematic for the Conv1D operator, since the `extend_attr_to_tuple` used here, will not work:
https://github.com/pytorch/TensorRT/blob/b3089bf00ec2e5124138201f6d76c8e3b6636b64/py/torch_tensorrt/fx/converters/impl/convolution.py#L111-L115

Specifically, the function `extend_attr_to_tuple`, shown below, needs to be modified to do the following. It should be able to handle length-1 lists or tuples and extend those to the necessary length specified by the user.
https://github.com/pytorch/TensorRT/blob/b3089bf00ec2e5124138201f6d76c8e3b6636b64/py/torch_tensorrt/fx/converters/converter_utils.py#L128-L146

## To Reproduce

Compile a model with a Conv1D operator using one of the Dynamo paths.

## Expected behavior

Models with Conv1D operators should trace and compile successfully with AOT/Dynamo.

## Environment

 - Torch-TensorRT Version (e.g. 1.0.0): 8c62fca7649ac2e8e74ab7d9d66a20a164e2af51
 - PyTorch Version (e.g. 1.0): `2.1.0.dev20230803+cu121`

**Comment by zewenli98** [comment]
> It should be able to handle length-1 lists or tuples and extend those to the necessary length specified by the user.

Sorry I don't get it. Could you give an example for this?

**Comment by gs-olive** [comment]
@zewenli98 - sure, for instance in the conv1d case, we can get inputs like `stride=[1]` and `dilation=[0]`. In those cases, the function `extend_attr_to_tuple` will not extend those lists because they are not single values. Specifically, `extend_attr_to_tuple([1], 2)` returns `[1]`, but we want it to return `[1, 1]`. It would help to have some logic in the `extend_attr_to_tuple` function which detects singleton lists/tuples and treats them like single elements. For instance:
```python
if isinstance(val, (tuple, list)) and len(val) == 1:
  val = val[0]
```

**Comment by zewenli98** [comment]
Thanks for the detailed explanation George! It's fixed in the PR #2252
